### ISSUE TITLE
Options for strict parsing?

### ISSUE THREAD

**Issue by paavohuhtala** [date-fns/date-fns#819]
I'm using `date-fns` (latest 2.0 alpha) to validate that certain datetime strings are in a specific ISO8601 compliant format. It's critically important that they are in the exact format I'm testing for so that certain invariants hold.

However, this parses as valid:

```
> dateFns.parse("2012-12-12T12:30", "YYYY-MM-dd", new Date())
2012-12-11T22:00:00.000Z
```

and even this:

```
> dateFns.parse("2012-12-12asfasfasfT12:30", "YYYY-MM-dd", new Date())
2012-12-11T22:00:00.000Z
```

I propose the parser should check if there's any remaining input after parsing, as long as a format was provided. Best-effort parsing is okay for UI use, but for critical validation tasks this is unwanted behaviour.

There should also be an option for disabling the `Date` constructor fallback, for the same reasons.

**Comment by leshakoss** [comment]
Fixed in [v2.0.0-alpha.16](
https://gist.github.com/kossnocorp/a307a464760b405bb78ef5020a4ab136#v200-alpha16)

**Comment by anasofiapaixao** [comment]
I am experiencing this issue with the parsing still. All of these parse as valid: 

``> dateFns.parse("2013-02-11T10:09:00", "YYYY-MM-DDTHH:mm:ss[Z]", new Date())

``> dateFns.parse("2013-02-11T10:09:", "YYYY-MM-DDTHH:mm:ss[Z]", new Date())

``> dateFns.parse("2013-02-11T10:0900", "YYYY-MM-DDTHH:mm:ss[Z]", new Date())

``> dateFns.parse("2013-22-11T10:09:00Z", "YYYY-MM-DDTHH:mm:ss[Z]", new Date())
